pkg/remote/client.go

```diff
@@ -89,8 +93,17 @@ func (c *WriteClient) Store(ctx context.Context, series []*prompb.TimeSeries) er
 	if err != nil {
 		return fmt.Errorf("HTTP POST request failed: %w", err)
 	}
-	defer resp.Body.Close()
-	io.Copy(io.Discard, resp.Body) //nolint:errcheck
+	defer func() {
+		err = resp.Body.Close()
+		if err != nil {
+			panic(err)
+		}
```

Comment:
It is too aggressive here, we have to expect errors here so we need to manage that better. I will do it directly.